popde/adaptive_kde.py

```diff
@@ -74,14 +74,43 @@ def set_per_point_bandwidth(self, pilot_values):
         """
         Calculate per-point bandwidths and re-initialize KDE
 
-        Parameters:
-        -----------
+        Parameters
+        ----------
         pilot_values : array-like
             The pilot KDE values at the data point positions.
         """
         self.set_bandwidth(
             self.global_bandwidth / self._local_bandwidth_factor(pilot_values)
         )
+    
+    def set_alpha(self, new_alpha):
+        """
+        Set the adaptive parameter alpha to a new value and re-initialize KDE.
+
+        Parameters
+        ----------
+        new_alpha : float
+            The new value for the adaptive parameter alpha.
+        """
+        self.alpha = new_alpha
+        # Update local bandwidths
+        self.set_per_point_bandwidth(self.pilot_values)       
+
+    def set_adaptive_parameter(self, new_alpha, new_global_bw):
+        """
+        Update the adaptive parameter alpha and the global bandwidth,
+        then reassign local bandwidths and re-initialize the KDE.
+
+        Parameters
+        ----------
+        new_alpha : float
+            The new value for the adaptive parameter alpha.
+        new_global_bw : float
+            The new value for the global bandwidth.
+        """
+        self.set_bandwidth(new_global_bw)
```

Comment:
Does this recalculate self.pilot_values ? As bw changes, the pilot values _do_ need to be recalculated here